cmd/main.go

```diff
@@ -67,16 +126,55 @@ func run() error {
 	return nil
 }
 
-func runScan(ctx context.Context, c *config.Config, client scan.EthClient, s scan.Scanner, storage scan.Storage) error {
-	var err error
-	startingBlock := c.StartingBlock
-	if startingBlock == 0 {
-		startingBlock, err = getL1LatestBlock(ctx, client)
+func compareChainIDs(ctx context.Context, client scan.EthClient, repositoryService repository.Service) error {
+	chainId, err := client.ChainID(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting chain ID from Ethereum client: %w", err)
+	}
+
+	chainIdDB, err := repositoryService.GetChainID()
+	if err != nil {
+		return fmt.Errorf("error getting chain ID from database: %w", err)
+	}
+
+	if chainIdDB == "" {
+		if err = repositoryService.SetChainID(chainId.String()); err != nil {
+			return fmt.Errorf("error setting chain ID in database: %w", err)
+		}
+	} else {
+		if chainId.String() != chainIdDB {
```

Comment:
You still can, you have to tell the node to use a different directory with `-storage_path`.

We decided to go with this after a brief discussion with the product owner. I initially proposed your solution, but we agreed to implement it the other way (what we currently have), in case we wanna support future non-EVM chains which don't have the concept of a chain id.

Although, while working on the solution, I noticed that the approach we were currently working on doesn't work with non-EVM chains either.

Either way, both approaches are fine. I'm totally open to accept your solution, but I'd kindly ask you to do it in another task.

Reply:
you could forget to set -storage_path and you could end up in dirty DB. So I still think that chainID in path is a better solution. But, I agree, lets do that in another task.